packages/flutter/lib/src/cupertino/scaffold.dart

```diff
@@ -113,9 +113,10 @@ class _CupertinoScaffoldState extends State<CupertinoScaffold> {
   /// Pad the given middle widget with or without top and bottom offsets depending
   /// on whether the middle widget should slide behind translucent bars.
   Widget _padMiddle(Widget middle) {
-    double topPadding = MediaQuery.of(context).padding.top;
+    double topPadding = 0.0;
     if (widget.navigationBar is CupertinoNavigationBar) {
```

Comment:
this is another example of depending on the widget type. We should never do that. It should always be possible to replace a widget with an identical but differently named widget and have the same result, or wrap a widget in another and have the same result.

Reply:
Will make an interface.